File: src/main/java/org/team1/nbe1_3_team01/domain/user/repository/CourseRepository.java
package org.team1.nbe1_3_team01.domain.user.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import org.team1.nbe1_3_team01.domain.user.entity.Course;

import java.util.Optional;

public interface CourseRepository extends JpaRepository<Course, Long> {
    Optional<Course> findByName(String name);
}

File: src/main/java/org/team1/nbe1_3_team01/domain/user/repository/CourseRepository.kt
package org.team1.nbe1_3_team01.domain.user.repository

import org.springframework.data.jpa.repository.JpaRepository
import org.team1.nbe1_3_team01.domain.user.entity.Course
import java.util.*

interface CourseRepository : JpaRepository<Course, Long> {
    fun findByName(name: String): Course?
}
